### PR TITLE
chore(ci): provide a basic theme workflow, add possibility to use a pre-built artifact for extension installation

### PR DESCRIPTION
**Pull request by philipreinken**
What's left to do on the theme side then:

```yaml
name: Theme
on:
  workflow_dispatch:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  ci:
    uses: shopware/github-actions/.github/workflows/theme.yml@main
```

### CODE REVIEW

File: .github/workflows/theme.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      - uses: actions/checkout@v4
+
+      - name: Determine whether to run Jest tests
+        id: should_run
+        run: |-
+          echo "test_jest=${{ hashFiles('Resources/app/storefront/package-lock.json') != '' && hashFiles('Resources/app/storefront/jest.config.*') != '' }}" >> ${GITHUB_OUTPUT}
+
+  test_jest:
+    name: Unit tests (Jest)
+    runs-on: ${{ github.event.act && 'ubuntu-latest' || format('runs-on={0}/runner=sw-amd64', github.run_id) }}
```
Comment (pweyck) [inline .github/workflows/theme.yml:68]: externals cannot use this like this. Maybe check private flag? or we could use an input argument?

File: .github/workflows/theme.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    outputs:
+      test_jest: ${{ steps.should_run.outputs.test_jest }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Determine whether to run Jest tests
+        id: should_run
+        run: |-
+          echo "test_jest=${{ hashFiles('Resources/app/storefront/package-lock.json') != '' && hashFiles('Resources/app/storefront/jest.config.*') != '' }}" >> ${GITHUB_OUTPUT}
```
Comment (pweyck) [inline .github/workflows/theme.yml:64]: I feel like for a single job there should be simpler solution.

Comment (philipreinken) [reply .github/workflows/theme.yml]: Yes, I think so, too. This workflow should be reusable, though, so we need to determine whether there are tests to be executed.

The `hashFiles` function is [unfortunately only available](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#context-availability) below the `jobs.<job_id>.steps` context and it is the only way to check for file availability directly within GHA.

So the alternative would be to add an `if: hashFiles(...)` to _every step_ of the test job, hence why I went with this approach.

File: .github/workflows/theme.yml (hunk 1 above)
Comment (philipreinken) [reply .github/workflows/theme.yml]: I've added a check for the `private` property of a repository. If it's false, we'll use the `ubuntu-latest` runner.

**Pull request merged**